src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs

```diff
@@ -231,29 +234,32 @@ internal static string ResolveContentRootPath(string contentRootPath, string bas
             return Path.Combine(Path.GetFullPath(basePath), contentRootPath);
         }
 
+        [MemberNotNull(nameof(_hostBuilderContext))]
         private void InitializeHostBuilderContext()
         {
             _hostBuilderContext = new HostBuilderContext(Properties)
             {
-                HostingEnvironment = _hostingEnvironment,
-                Configuration = _hostConfiguration
+                HostingEnvironment = _hostingEnvironment!, // TODO-NULLABLE: https://github.com/dotnet/csharplang/discussions/5778
```

Comment:
I think one of these comments is "good enough". At the end of the comment above you can just mention "and similarly below" at the end. I don't think we need to clutter our code all over with these TODOs for a request that may not make it into the product.